sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -578,6 +598,8 @@ def read(self, size: Optional[int] = -1) -> T:
             the return value is empty, there is no more data to read.
         :rtype: T
         """
+        if size is None:
+            return self.readall()
```

Comment:
Ah, I think I messed up the type hint for `size` here. It should be just `int` not `Optional`. If you could fix that, then you wouldn't need this.